File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.java
class Solution {
    public int maximumElementAfterDecrementingAndRearranging(int[] arr) {
        Arrays.sort(arr);
        int ans = 1;
        arr[0] = 1;
        for(int i=1; i<arr.length; i++) {
            if(arr[i]-arr[i-1] > 1) {
                arr[i] = arr[i-1] + 1;
            }
            ans = Math.max(ans, arr[i]);
        }
        return ans;
    }
}

File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.js
/**
 * @param {number[]} arr
 * @return {number}
 */
var maximumElementAfterDecrementingAndRearranging = function(arr) {
    const n = arr.length;
    const ar = Array(n+1).fill(0);
    for(let num of arr) {
        ar[Math.min(num, n)]++;
    }
    let ans = 1;
    for(let i=1; i<=n; i++) {
        ans = Math.min(i, ans + ar[i]);
    }
    return ans;
};